msteams-platform/task-modules-and-cards/task-modules/invoking-task-modules.md

```diff
@@ -0,0 +1,236 @@
+| `card` | Adaptive Card or Adaptive Card bot card attachment | This attribute is the JSON for the Adaptive Card to appear in the task module. If the user is invoking from a bot, use the Adaptive Card JSON in a Bot Framework `attachment` object. From a tab, the user must use an Adaptive Card. For more information, see [Adaptive Card or Adaptive Card bot card attachment](#adaptive-card-or-adaptive-card-bot-card-attachment) |
+| `fallbackUrl` | string | This attribute opens the URL in a browser tab, if a client does not support the task module feature. |
+| `completionBotId` | string | This attribute specifies a bot App ID to send the result of the user's interaction with the task module. If specified, the bot receives a `task/submit invoke` event with a JSON object in the event payload. |
+
+> [!NOTE]
+> The task module feature requires that the domains of any URLs you want to load are included in the `validDomains` array in your app's manifest.
+
+The next section specifies task module sizing that enables the user to set the height and width of the task module.
+
+## Task module sizing
+
+Using integers for `TaskInfo.width` and `TaskInfo.height`, sets the height and width of the task module in pixels. However, depending on the size of the Team's window and screen resolution they are reduced proportionally while maintaining the aspect ratio that is width or height.
+
+If `TaskInfo.width` and `TaskInfo.height` are `"small"`, `"medium"`, or `"large"`, the size of the red rectangle in the following image is a proportion of the available space, 20%, 50%, and 60% for `width` and 20%, 50%, and 66% for `height`:
+
+![Task module example](~/assets/images/task-module/task-module-example.png)
+
+Task modules invoked from a tab can be dynamically resized. After calling `tasks.startTask()` you can call `tasks.updateTask(newSize)` where height and width properties on the newSize object conform to the TaskInfo specification, for example `{ height: 'medium', width: 'medium' }`.
+
+The next section provides examples of embedding task modules in a YouTube video and a PowerApp.
+
+## Task module CSS for HTML or JavaScript task modules
+
+HTML or JavaScript-based task modules have access to the entire area of the task module below the header. While that offers a great deal of flexibility, if you want padding around the edges to align with the header elements and avoid unnecessary scroll bars, the user must provide the right CSS. The next sections provide some examples for a few use cases.
+
+**Example 1: YouTube video**
+
+YouTube offers the ability to embed videos on web pages. It is easy to embed videos on web pages in a task module using a simple stub web page.
+
+![YouTube video](~/assets/images/task-module/youtube-example.png)
+
+The following code provides an example of the HTML for the web page without the CSS:
```

Comment:
Give location to access this code. What can you do with this code? Please explain.

Reply:
Awaiting inputs from Wajeed